agent/build.gradle

```diff
@@ -1,88 +1,150 @@
-            from(zipTree(it)) {
-                into("inst")
-                rename("(^.*)\\.class\$", "\$1.classdata")
-            }
-        }
+  return copySpec {
+    configurations.customShadow.files.each {
+      from(zipTree(it)) {
+        into("inst")
+        rename("(^.*)\\.class\$", "\$1.classdata")
+      }
     }
+  }
 }
 
 // includes everything except otel agent
 task customShadow(type: ShadowJar) {
-    from sourceSets.main.output
+  from sourceSets.main.output
 
-    configurations = [project.configurations.customShadowInclude]
+  configurations = [project.configurations.customShadowInclude]
 
-    dependsOn ':custom:shadowJar'
-    dependsOn ':instrumentation:shadowJar'
-    dependsOn ':profiler:shadowJar'
-    dependsOn ':bootstrap:jar'
-    dependsOn ':agent:jar'
-    with isolateSpec()
+  dependsOn ':custom:shadowJar'
+  dependsOn ':instrumentation:shadowJar'
+  dependsOn ':profiler:shadowJar'
+  dependsOn ':bootstrap:jar'
+  dependsOn ':agent:jar'
+  with isolateSpec()
 
-    mergeServiceFiles()
-    exclude("**/module-info.class")
+  mergeServiceFiles()
+  exclude("**/module-info.class")
 
-    relocatePackages(it)
+  relocatePackages(it)
 }
 
 tasks {
-    // merge otel agent with our agent
-    shadowJar {
-        from project.configurations.mainShadow.files
-        from tasks.customShadow.outputs
+  // merge otel agent with our agent
+  shadowJar {
+    archiveClassifier.set("all")
 
-        mergeServiceFiles {
-            include("inst/META-INF/services/*")
-        }
+    from project.configurations.mainShadow.files
+    from tasks.customShadow.outputs
 
-        manifest {
-            attributes.put("Main-Class", "io.opentelemetry.javaagent.OpenTelemetryAgent")
-            attributes.put("Agent-Class", "com.splunk.opentelemetry.SplunkAgent")
-            attributes.put("Premain-Class", "com.splunk.opentelemetry.SplunkAgent")
-            attributes.put("Can-Redefine-Classes", "true")
-            attributes.put("Can-Retransform-Classes", "true")
-            attributes.put("Implementation-Vendor", "Splunk")
-            attributes.put("Implementation-Version", "splunk-${project.version}-otel-${versions["opentelemetryJavaagent"]}")
-        }
+    mergeServiceFiles {
+      include("inst/META-INF/services/*")
+    }
+
+    manifest {
+      attributes.put("Main-Class", "io.opentelemetry.javaagent.OpenTelemetryAgent")
+      attributes.put("Agent-Class", "com.splunk.opentelemetry.SplunkAgent")
+      attributes.put("Premain-Class", "com.splunk.opentelemetry.SplunkAgent")
+      attributes.put("Can-Redefine-Classes", "true")
+      attributes.put("Can-Retransform-Classes", "true")
+      attributes.put("Implementation-Vendor", "Splunk")
+      attributes.put("Implementation-Version", "splunk-${project.version}-otel-${versions["opentelemetryJavaagent"]}")
     }
+  }
+
+  assemble {
+    dependsOn(shadowJar)
+  }
+}
+
+rootProject.tasks.named("release").configure {
+  finalizedBy(tasks["publishToSonatype"])
+}
```

Comment:
It's groovy, does not `release.finalizedBy publishToSonatype` work?

Reply:
Yeah, it'd work, but I don't want to add more work for the Kotlin migration (coming soon)